apps/web/src/hooks/useThirdYearBirthdayEffect.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useCallback } from 'react'
+import { useRouter } from 'next/router'
+import confetti from 'canvas-confetti'
+import debounce from 'lodash/debounce'
+
+const defaultOptions = {
+  debounceDuration: 200,
+}
+
+type Options = {
+  selector?: string
+  debounceDuration?: number
+  disableWhen?: () => boolean
+}
+
+const useConfetti = (options: Options): { initialize: () => void; teardown: () => void } => {
+  const { selector, debounceDuration, disableWhen } = {
+    ...defaultOptions,
+    ...options,
+  }
+
+  const makeListener = useCallback(
+    () =>
+      debounce(
+        () => {
+          const isDisabled = disableWhen && disableWhen()
+
+          if (!isDisabled) {
+            confetti({
+              particleCount: 100,
+              startVelocity: 30,
+              spread: 360,
+              origin: {
+                x: Math.random(),
+                y: Math.random() - 0.2,
+              },
+            })
+          }
+        },
+        debounceDuration,
+        { leading: true },
+      ),
+    [debounceDuration, disableWhen],
+  )
+  const listener = makeListener()
```

Comment:
This will create a new listener every time this hook is run?

Reply:
I think we can use `useMemo` instead of `useCallback`?

Reply:
done